reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -284,14 +284,14 @@ default Context putAll(ContextView other) {
 	}
 
 	/**
-	 * See {@link #putAll(ContextView)}.
+	 * Convenience alternative to {@link #putAll(ContextView)} where one wants to specify several key-value pairs,
+	 * since a common way of providing such pairs is via {@link Context#of(Object, Object, Object, Object, Object, Object, Object, Object, Object, Object)}.
+	 * This method simply calls the {@link Context#readOnly()}, which generally casts the {@link Context} as a {@link ContextView}.
```

Comment:
This is phrased a bit ambiguously I think. It's a convenience alternative to `putAll(ContextView)` not because you want to specify multiple key-value pairs (in that regard they're identical) but because of the downcast.

I suggest to put the verbiage about one wanting to specify several key-value pairs on `putAll(ContextView)` and then here to say something like:
```
Convenience alternative to {@link #putAll(ContextView)} that accepts a `Context` created through any of the Context#of` methods without the need to downcast. 
```
Then again with such a description it raises the question, does this method even needs to exist in the long run? I can see also where the deprecation notice or the Javadoc here could suggest use of `Context#of` and then `.readOnly()` as a temporary workaround.

Reply:
> Then again with such a description it raises the question, does this method even need to exist in the long run?

yeah, it confirms the original intent, kind of.

with de8db50 I thus turned this PR into one that maintains the deprecation but gives pointer to `readOnly()`